{
  try {
    if (cleanupAfterMethod())     setup();
  }
 catch (  Exception e) {
    log.error("Unexpected!",e);
    throw e;
  }
}
